Source/XR-Interaction-Component/Source/Runtime/Interaction/Interactors/SnapZone.cs

```diff
@@ -354,6 +354,20 @@ protected virtual void UpdateHighlightMeshFilterCache()
             if (meshes.Any())
             {
                 previewMesh = new Mesh();
+                int totalVertices = 0;
+
+                for (int i = 0; i < meshes.Count; ++i)
+                {
+                    totalVertices += meshes[i].mesh.vertexCount;
+                }
+
+                // If the total vertices of the combined mesh would be above the maximum,
+                // change the index format of the preview mesh to int32 to avoid errors.
+                if (totalVertices > 65534)
+                {
+                    previewMesh.indexFormat = IndexFormat.UInt32;
+                }
+
                 previewMesh.CombineMeshes(meshes.ToArray());
                 previewMesh.UploadMeshData(true);
```

Comment:
Suggesting to move the whole content in to a separate method.
`previewMesh = CreatePreviewMesh(List<CombineInstance> meshes)`